nupic/research/monitor_mixin/plot.py

```diff
@@ -46,40 +49,70 @@ def __init__(self, monitor, title):
     plt.show()
 
 
+
+  def _initFigure(self):
+    fig = plt.figure()
+    fig.suptitle(self._prettyPrintTitle())
+    return fig
+
+
+
+  def _prettyPrintTitle(self):
+    if self._monitor.mmName is not None:
+      return "[{0}] {1}".format(self._monitor.mmName, self._title)
+    return self._title
+
+
+
   def addGraph(self, data, position=111, xlabel=None, ylabel=None):
     ax = self._addBase(position, xlabel=xlabel, ylabel=ylabel)
     ax.plot(data)
     plt.draw()
 
 
-  def addHistogram(self, data, position=111, xlabel=None, ylabel=None,
-                   bins=None):
+
+  def addImage(self, data, position=111, xlabel=None, ylabel=None,
```

Comment:
Since this calls imshow() I deferred to the documentation for that function (http://matplotlib.org/api/axes_api.html ) which says "Display an image on the axes." What addImageToAxes()? add2DArray()?